lib/compress/zstd_compress.c

```diff
@@ -2513,7 +2515,7 @@ static size_t ZSTD_compressBlock_targetCBlockSize(ZSTD_CCtx* zc,
         }
     }
 
-    if (!ZSTD_isError(cSize) && !usingNoCompressSuperBlock) {
+    if (!ZSTD_isError(cSize) && compressSuperBlock) {
```

Comment:
> However, when (!usingNoCompressSuperBlock), there is still a possibility that a compress block was sent. It's just no longer divided into sub-blocks.

I think this was a mistake. Now we should only enter this branch if the block sent is compressed. And this should take care of the rep codes too I believe.